lib/java/com/google/android/material/internal/CollapsingTextHelper.java

```diff
@@ -593,19 +590,15 @@ private void calculateBaseOffsets() {
             isRtl ? ViewCompat.LAYOUT_DIRECTION_RTL : ViewCompat.LAYOUT_DIRECTION_LTR);
     switch (expandedAbsGravity & Gravity.VERTICAL_GRAVITY_MASK) {
       case Gravity.BOTTOM:
-        float offset = shouldDrawMultiline() ? textLayoutHeight - textPaint.descent() : 0;
-        expandedDrawY = expandedBounds.bottom - offset;
+        expandedDrawY = expandedBounds.bottom - expandedTextHeight;
```

Comment:
These changes are for `expandedDrawY` and `collapsedDrawY ` to always mark top edge of the text to draw. Before, it was top edge for multiline, but bottom (baseline actually) for single line.